src/webots/nodes/WbSkin.cpp

```diff
@@ -728,7 +728,8 @@ bool WbSkin::createSkeletonFromWebotsNodes() {
       const WbVector3 unit(0, 0, 1);
       const WbVector3 &norm = offset.normalized();
       const WbVector3 &axis = unit.cross(norm).normalized();
-      const float boneOrientation[4] = {(float)unit.angle(norm), (float)axis[0], (float)axis[1], (float)axis[2]};
+      const float angle = offset.length() ? (float)unit.angle(norm) : 0;
```

Comment:
I think it would be better to move the check in the `WbVector3::angle` function directory to avoid similar issues in the future.
Moreover we do not need to compute the sqrt but  `WbVector3::length2` could be checked instead.